extensions/notebook/src/jupyter/serverInstance.ts

```diff
@@ -207,12 +210,36 @@ export class PerNotebookServerInstance implements IServerInstance {
 	}
 
 	private async copyKernelsToSystemJupyterDirs(): Promise<void> {
+		const DarwinPostfix = '_darwin';
+		const LinuxPostfix = '_linux';
+		const Win32Postfix = '_win32';
 		let kernelsExtensionSource = path.join(this.options.install.extensionPath, 'kernels');
 		this._systemJupyterDir = path.join(this.getSystemJupyterHomeDir(), 'kernels');
 		if (!(await this.utils.exists(this._systemJupyterDir))) {
 			await this.utils.mkDir(this._systemJupyterDir, this.options.install.outputChannel);
 		}
-		await this.utils.copy(kernelsExtensionSource, this._systemJupyterDir);
+		let platform = process.platform;
+		let subFolders = await this.utils.readDir(kernelsExtensionSource);
+		if (platform === 'win32') {
+			subFolders = subFolders.filter(folder => !folder.includes(DarwinPostfix) && !folder.includes(LinuxPostfix));
+		} else if (platform === 'darwin') {
+			subFolders = subFolders.filter(folder => !folder.includes(LinuxPostfix) && !folder.includes(Win32Postfix));
+		} else {
+			subFolders = subFolders.filter(folder => !folder.includes(DarwinPostfix) && !folder.includes(Win32Postfix));
+		}
+		for (let i = 0; i < subFolders.length; i++) {
+			let realFolder = subFolders[i];
+			// Ignore any system files
+			if (!realFolder.startsWith('.')) {
+				if (realFolder.includes(process.platform)) {
+					let splitFolder = realFolder.split('_' + process.platform);
+					if (splitFolder && splitFolder.length > 0) {
+						realFolder = splitFolder[0];
+					}
+				}
+				await this.utils.copy(path.join(kernelsExtensionSource, subFolders[i]), path.join(this._systemJupyterDir, realFolder));
```

Comment:
What happens if this fails?

Reply:
Good question. An error message pops up to users here. Any other suggestions?

Reply:
Ah as long as its handled somewhere :) Sometimes these failures completely break the extension so just checking :)